features/step_definitions/logging.rb

```diff
@@ -1056,3 +1056,139 @@
+
+    template_file = "#{file_dir}/container_json_log_template.json"
+    if log_type == "unicode" || log_type == "flat"
+      template_file = "#{file_dir}/container_json_unicode_log_template.json"
+    end
+
+    unless project(project_name).exists?
+      step %Q/I run the :new_project client command with:/,table(%{
+        | project_name | #{project_name} |
+      })
+      raise "Error creating namespace" unless @result[:success]
+    end
+
+    step %Q/I use the "#{project_name}" project/
+
+    if log_type == "any"
+      pods = project.pods(by:user)
+      pods.each do | current_pod |
+        cache_pods(current_pod)
+        @result = user.cli_exec(:logs, {resource_name: "#{current_pod.name}", n: "#{project_name}", since: "15s"})
+        if @result[:success]
+           deploy_pod = false
+        end
+      end
+    end
+
+    if deploy_pod
+      step %Q/I run the :delete client command with:/, table(%{
+        | object_type | configmap |
+        | object_name_or_id | logtest-config |
+      })
+      step %Q/I run the :delete client command with:/, table(%{
+        | object_type | ReplicationController |
+        | object_name_or_id | centos-logtest |
+      })
+      step %Q/I run the :new_app client command with:/,table(%{
+        | file | #{template_file} |
+      })
+    end
+end
+
+Given /^I have index pattern #{QUOTED}$/ do | pattern_name |
+    step %Q/I perform the :kibana_index_pattern_exist web action with:/,table(%{
+      | index_pattern_name | #{pattern_name} |
+    })
+    unless @result[:success]
+      step %Q/I run the :go_to_kibana_management_page web action/
+      step %Q/I perform the :create_index_pattern_in_kibana web action with:/, table(%{
+        | index_pattern_name | "#{pattern_name}" |
+       })
+       raise "Failed to create index pattern #{pattern_name}" unless @result[:success]
+    end
+end
+
+Given /^cluster-admin create the #{QUOTED} pattern for the #{WORD} user$/ do | pattern_name, who |
+    user(word_to_num(who))
+    user_name=user.name
+    user_token=user.cached_tokens.first
+    step %Q/I switch to cluster admin pseudo user/
+    step %Q/I use the "openshift-logging" project/
+    
+    rest_cmd="curl -s --connect-timeout 10  -XPOST \"http://localhost:5601/api/saved_objects/index-pattern/#{pattern_name}\" -H \"kbn-xsrf: true\" -H \"x-forwarded-user: #{user_name}\" -H \"securitytenant: __user__\" -H \"Authorization: Bearer #{user_token}\" -H \"Content-Type: application/json\" -d \'{\"attributes\": {\"title\": \"#{pattern_name}\", \"timeFieldName\": \"@timestamp\"}}\'"
+
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | component= kibana|
+    })
+    @result = pod.exec("bash", "-c", rest_cmd, as: admin, container: 'kibana')
+    raise "Failed to create index pattern via rest api" unless @result[:success]
+end
+
+Given /^I can display the(?: "(.+)")? pod logs of the(?: "(.+)")? project under the(?: "(.+)")? pattern in kibana$/ do | pod_name, project_name, pattern_name |
```

Comment:
@anpingli if you looks at [_constants.rb](https://github.com/openshift/verification-tests/blob/master/features/step_definitions/_constants.rb) file, there contains some useful alias that you can use in the step definition.